core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractProcessingStrategy.java

```diff
@@ -55,7 +53,7 @@ public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor pipeline)
     };
   }
 
-  protected ExecutorService decorateScheduler(Scheduler scheduler) {
+  protected ScheduledExecutorService decorateScheduler(ScheduledExecutorService scheduler) {
```

Comment:
can the param and return type be `ExecutorService` (not `Scheduled*`)? Or does that change break something?